sycl/test-e2e/Matrix/joint_matrix_hip_apply.hpp

```diff
@@ -0,0 +1,102 @@
+
+#include <sycl/sycl.hpp>
+
+#include <cstddef>
+#include <cstdint>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using sycl::ext::oneapi::bfloat16;
+
+template <typename InType, typename OutType, size_t M, size_t N, size_t K>
+void hip_matrix_mfma() {
+  InType A[M * K];
+  InType B[K * N];
+  OutType C[M * N];
+  OutType D[M * N];
+  OutType E[M * N];
+
+  for (auto i = 0; i < M * K; ++i) {
+    A[i] = 1;
+  }
+
+  for (auto i = 0; i < K * N; ++i) {
+    B[i] = 2;
+  }
+
+  for (auto i = 0; i < M * N; ++i) {
+    D[i] = 0;
+    C[i] = 3;
+    E[i] = 3;
+  }
+
+  try {
+    auto defaultQueue = sycl::queue{};
+
+    auto bufA = sycl::buffer{A, sycl::range{M * K}};
+    auto bufB = sycl::buffer{B, sycl::range{K * N}};
+    auto bufC = sycl::buffer{C, sycl::range{M * N}};
+    auto bufD = sycl::buffer{D, sycl::range{M * N}};
+
+    defaultQueue
+        .submit([&](sycl::handler &cgh) {
+          sycl::accessor accA{bufA, cgh, sycl::read_write};
+          sycl::accessor accB{bufB, cgh, sycl::read_write};
+          sycl::accessor accC{bufC, cgh, sycl::read_only};
+          sycl::accessor accD{bufD, cgh, sycl::write_only};
+
+          cgh.parallel_for(
+              sycl::nd_range<2>{{4, 16}, {4, 16}}, [=](sycl::nd_item<2> idx) {
+                auto sg = idx.get_sub_group();
+                joint_matrix<sub_group, OutType, use::accumulator, M, N>
+                    sub_c{};
+                joint_matrix<sub_group, InType, use::b, K, N, layout::row_major>
+                    sub_b{};
+                joint_matrix<sub_group, InType, use::a, M, K, layout::col_major>
+                    sub_a{};
+
+                joint_matrix_load(
+                    sg, sub_a,
+                    accA.template get_multi_ptr<access::decorated::yes>(), K);
```

Comment:
Using `access::decorated::yes` is unecessary and just adds confusion/complexity.

```suggestion
                    accA.template get_multi_ptr(), K);
```

Reply:
Thanks @JackAKirk 
They are all changed.

Reply:
@mmoadeli , I still see access::decorated::yes used, so it seems, this was not yet addressed.

Reply:
> @mmoadeli , I still see access::decorated::yes used, so it seems, this was not yet addressed.

@YuriPlyakhin `get_multi_ptr` does not have a default template parameter for decoration type. Removing results in compilation error. The default value for decoration type was `legacy` which is going to be deprecated. So, we need to have `access::decorated`. I ran it offline with @JackAKirk